juju/juju
# Extract payloads apiserver parts

## Description of change

This PR moves the apiserver components of the "payloads" feature from the top level `payload` package to under the `apiserver` package where the rest of the API is defined. It is part of a larger effort to clean up the apiserver in preparation for a major new feature.

## QA steps

There should be no externally visible changes. In order to check that a regression hasn't been introduced a payload using charm was deployed:

```
$ juju bootstrap lxd
$ juju deploy ~/canonical/repository/payloadsfoo
# wait until it deploys
$ juju list-payloads
[Unit Payloads]
Unit            Machine  Payload class  Status   Type    Id   Tags                                   
payloadsfoo/0   1        foo            running  mocker  Wed  Mar 8 19:28:26 UTC 2017
```

This ensures that both the unit facade (which is used to set payloads) and the payloads facade (which is used to retrieve payloads) are working.

## Documentation changes

No externally visible changes.

## Bug reference

N.A.
